src/main/java/com/uber/cadence/internal/testservice/TestWorkflowMutableStateImpl.java

```diff
@@ -133,7 +133,7 @@ void apply(RequestContext ctx)
   private final Map<String, StateMachine<TimerData>> timers = new HashMap<>();
   private final Map<String, StateMachine<SignalExternalData>> externalSignals = new HashMap<>();
   private StateMachine<WorkflowData> workflow;
-  private StateMachine<DecisionTaskData> decision;
+  private volatile StateMachine<DecisionTaskData> decision;
```

Comment:
May be reuse existing lock?

Reply:
The existing lock seems to protect this. I'll revert this change.